can/interfaces/vector/canlib.py

```diff
@@ -93,6 +97,25 @@ def __init__(self, channel, can_filters=None, poll_interval=0.01,
         # Get channels masks
         self.channel_masks = {}
         self.index_to_channel = {}
+
+        if serialNumber is not None:
+            app_name = None
+            globalChannel = []
+            hwSpecificChannel = 0
+            hardwareConfigs = self._detect_available_configs()
+            for hardwareConfig in hardwareConfigs:
+                if hardwareConfig['serialNumber'] == serialNumber:
+                    if hwSpecificChannel in self.channels:
+                        globalChannel.append(hardwareConfig['channel'])
+                    hwSpecificChannel = hwSpecificChannel + 1
```

Comment:
Could be safer to use `XLchannelConfig.hwChannel` since I don't think it's explicitly stated that the hardware configs will come in a certain order.